cdap-test/src/main/java/co/cask/cdap/test/AbstractApplicationManager.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.test;
+
+import co.cask.cdap.proto.Id;
+import com.google.common.base.Stopwatch;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Instance of this class is for managing deployed application.
+ */
+public abstract class AbstractApplicationManager implements ApplicationManager {
+
+  protected abstract boolean isRunning(Id.Program programId);
+
+  protected void programWaitForFinish(long timeout, TimeUnit timeoutUnit,
```

Comment:
the program prefix is just to avoid conflicting with the method names on the program manager vs the method names on the application managers. I tried removing it, and even though the signature is different, there's conflicts.